docs/templates/docker-in-workspaces.md

```diff
@@ -136,6 +136,40 @@ to sysbox's [compatibility matrix](https://github.com/nestybox/sysbox/blob/maste
 
 To get started with `envbox` check out the [starter template](../../examples/templates/envbox) or visit the [repo](https://github.com/coder/envbox).
 
+### Authenticating with a Private Registry
+
+Authenticating with a private container registry can be done by referencing the credentials
+via the `CODER_IMAGE_PULL_SECRET` environment variable. It is encouraged to populate this
+[environment variable](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#define-container-environment-variables-using-secret-data) by using a Kubernetes [secret](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials).
+
+Refer to your container registry documentation to understand how to best create this secret.
+
+The following shows a minimal example using a the JSON API key from a GCP service account to pull
+a private image:
+
+```bash
+# Create the secret
+$ kubectl create secret docker-registry <name> \
+  --docker-server=us.gcr.io \
+  --docker-username=_json_key \
+  --docker-password="$(cat ./json-key-file.yaml)" \
+  --docker-email=<service-account-email>
+```
+
+```yaml
+# An example of referencing a secret in an environment variable.
+kind: Pod
+spec:
+  containers:
+    - name: envbox
+      env:
+        - name: CODER_IMAGE_PULL_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: <name>
+              key: .dockerconfigjson
+```
```

Comment:
It might make sense to show this in Terraform instead of YAML since 99% of people on this doc will be looking to add this to their envbox template in Coder. Or show both 🤷🏼 

```hcl
env {
  name = "CODER_IMAGE_PULL_SECRET"
  value_from {
    secret_key_ref {
      name = "<name>"
      key = ".dockerconfigjson"
    }
  }
}
```